### PR TITLE
Feature/direct replies mute logic

### PR DESCRIPTION
**Pull request by johnxnguyen**
## What's new in this PR?

This PR introduces changes for direct replies in the context of muted conversation logic:
- `MutedMessageTypes` options have been renamed to reflect the inclusion of direct replies. The logic here hasn't changed, since replies are treated exactly the same as mentions. I've renamed `nonMentions` to `regular` and `mentions` to `mentionsAndReplies`. The way to understand this is `regular` means messages that are neither a self mention nor a self reply are muted.
- The data model has been updated to include an `internalEstimatedUnreadSelfReplyCount`.
- The unread conversation predicate has been updated to use this new property.
- The unarchiving behaviour has been updated to consider replies to the self user.
- The `ZMConversationListIndicator` has been updated to include a case for unread self replies.

### CODE REVIEW

File: Source/Model/Conversation/ZMConversation+Internal.h [hunk 1]
```diff
@@ -73,6 +73,7 @@ extern NSTimeInterval ZMConversationDefaultLastReadTimestampSaveDelay;
 extern NSString *const ZMConversationEstimatedUnreadCountKey;
 
 extern NSString *const ZMConversationInternalEstimatedUnreadSelfMentionCountKey;
+extern NSString *const ZMConversationInternalEstimatedUnreadSelfReplyCountKey;
```
Comment (billypchan) [inline Source/Model/Conversation/ZMConversation+Internal.h]: I think "Self" is not necessary and just name it "ZMConversationInternalEstimatedUnreadReplyCountKey" ?

Comment (mikeger) [reply Source/Model/Conversation/ZMConversation+Internal.h]: I think Self is included to underline that only the mentions of self user are counted. So I think it is valid.

Comment (johnxnguyen) [reply Source/Model/Conversation/ZMConversation+Internal.h]: Also true 👍

File: Source/Model/Conversation/ZMConversation+Mute.swift [hunk 2]
```diff
@@ -45,14 +46,14 @@ public struct MutedMessageTypes: OptionSet {
...
-    public static let all: MutedMessageTypes = [.nonMentions, .mentions]
+    public static let all: MutedMessageTypes = [.regular, .mentionsAndReplies]
     
-    /// Only non-mentions are muted.
-    public static let nonMentions = MutedMessageTypes(rawValue: MutedMessageOptionValue.nonMentions.rawValue)
+    /// Only regular messages (no mentions nor replies) are muted.
+    public static let regular = MutedMessageTypes(rawValue: MutedMessageOptionValue.regular.rawValue)
     
-    /// Only mentions are muted. Only used to check the bits in the bitmask.
+    /// Only mentions and replis are muted. Only used to check the bits in the bitmask.
```
Comment (mikeger) [inline Source/Model/Conversation/ZMConversation+Mute.swift]: Typo: `replis`

**Pull request merged**